### ISSUE TITLE
Path Unification for better merging

### ISSUE THREAD

**Issue by phanlezz** [MaibornWolff/codecharta#3495]
## Path Unification

> modules as in _importer_ / _parser_

All modules of the analysis should produces similar paths if possible for the same code. 
As a user I want to merge a variety of used modules, without modifying paths/roots of files. 

The motivation for this issue was the sonar importer, which unfortunately currently nests the paths additionally, but it should be checked to what extent this also affects the other modules. 

One approach could be to develop an automatic merge mode that tries to find a lowest common denominator across multiple project files. This mode should allow paths to be matched and folders to be nested correctly without manual input. 

A short example:

```
- root
- - a
- - b

- someKey
- - root
- - a
- - b

- b 

should all get merged without user input to:

- root
- - a
- - b
```

## Open questions

Modify modules or automatic merge

## Acceptance Criteria
All parsers and importers have the same output structure:
- [x] CodeMaat
- [x] CSV
- [x] MG
- [x] Sonar
    - Fixed by removing top level folder
- [x] Tokei
- [x] SourceMonitor
- [x] GitLog
- [x] RawText
- [x] SourceCode
- [x] SVN

> unification might be the wrong term.

**Comment by Richargh** [comment]
> The motivation for this issue was the sonar importer, which unfortunately currently nests the paths additionally

If that is the main motivation, wouldn't it then be better to work on the sonar importer? If sonar returns a single root folder and that contains the other stuff:

```
- root
- - theprojectname
- - - src/
- - - something.else
```

then instead return:

```
- root
- - src/
- - something.else
```

If sonar instead returns 

```
- root
- - anotherfolder
- - theprojectname
- - - src/
- - - something.else
```

Then return that because there must be some reason. 

Wouldn't that solve most problems and be the easiest to implement? 

PS: The tokeiimporter has the same problem and could be fixed in a similar way. `tokei codecharta -o json > codecharta.tokei.json` and `ccsh tokeiimporter codecharta.tokei.json -o codecharta.tokei.cc.json.gz` will return a file with the contents:

```
root
- codecharta
- - analysis
- - gh-pages
- - visualization
- - package.json
```

But it would be much better if it returned:

```
root
- analysis
- gh-pages
- visualization
- package.json
```

**Comment by phanlezz** [comment]
I agree, that the answer to the question, that I asked, is to modify the existing parsers to produce similar structures. After that, we can look into the topic of automatic merging, and if it is something useful to have as a tool.

**Comment by BenediktMehl** [comment]
> PS: The tokeiimporter has the same problem and could be fixed in a similar way. `tokei codecharta -o json > codecharta.tokei.json` and `ccsh tokeiimporter codecharta.tokei.json -o codecharta.tokei.cc.json.gz` will return a file with the contents:

The tokeiImporter has a Parameter "r", that needs to be specified to ensure the structure matches other parsers. If you try this command, it should work:
`ccsh tokeiimporter codecharta.tokei.json -o codecharta.tokei.cc.json.gz -r codecharta`
